README.md

```diff
@@ -1,26 +1,81 @@
 # What is this?
-It's a port of the Godot open source game engine to the Switch, via homebrew libraries provided by devkitPro/switchbrew.
+
+It's a port of the Godot open source game engine to the Switch, via homebrew libraries provided by `devkitPro` /`switchbrew`.
+
 See the original README-original.md for info about Godot.
 
-# Releases
+## Releases
+
 See [the GitHub releases page](https://github.com/Stary2001/godot/releases) for pre-built editors for Windows, macOS and Linux.
 
-# How do I export my game?
-The editor builds above add a `Switch` exporter to the list - it will generate a .nro and a .pck file that can be transferred to your Switch.
-If additional debugging is required, sending the nro via nxlink (make sure to send it to the right place with the -p argument) will allow the output of the console to be viewed on PC.
+## How do I export my game?
+
+The editor builds above add a `Switch` exporter to the list - it will generate a `.nro` and a `.pck` file that can be transferred to your Switch.
+
+Both file `.nro` and `.pck` nned to be "placed" in thte `switch` folder of you switch. Use your preferred method it could be directly mount your SD in your PC, using ftp app.
+
+## How do I map my controller?
+
+Up to 4 controllers can be used on the switch, they'll match the controller configuration done in the switch parameters.
+
+Controller names is the only "link" we have to tell godot what is the controller switch style. So there is a little bit of semantic in it for the one that need it. It goes as follow: `switch-pad-<id>::<style>::#color`.
+
+For example firt controller is joy-con in dual node vertical it will be `switch-pad-0::dual::#ffe6b90a::#ff283cff` (yes there is 2 colors because left and righ con have both there color). Another example second controller is a left joy-con in horizontal mode ti will be `switch-pad-0::solo-left::#ffe6b90a` and so on.
+
+**Vertical setup** `dual`/`pro`/`handheld` is supported with a pretty generic mapping, nothing special. (this mode also provide left anr right sticks analog axis).
+
+**Horizontal setup** `solo-left`/`solo-right` is also supported with funkier mapping. To provide consistency) the vertical style provide re-oriented D-PAD + 4 buttons + L/R. (No stick axis for this configuration --> fake D-PAD).
+
+Note: pad "style" is detected at first button down on a controller and is not re-check once the game is started.
+
+### How to debug controller?
```

Comment:
Stuff like this should probably get its own documentation page, if we can host a copy of the Godot docs. It doesn't really need to be in the README.

Reply:
Ok.